.changesets/fix_bnjjj_fix_4348.md

```diff
@@ -0,0 +1,5 @@
+### Fix fragment spread response formatting with aliased typename ([Issue #4348](https://github.com/apollographql/router/issues/4348))
+
+When you had an aliased `__typename` (example `myAlias: __typename`) in a fragment spread, the response formatting was wrong.
```

Comment:
the changeset should probably reflect the actual issue, which is that response formatting was confused by fragments on unions